pkg/kubectl/describe.go

```diff
@@ -1779,10 +1779,53 @@ func (d *ServiceAccountDescriber) Describe(namespace, name string, describerSett
 		}
 	}
 
-	return describeServiceAccount(serviceAccount, tokens)
+	existingPullSecrets := map[string]api.Secret{}
```

Comment:
I don't think pull secrets are limited to opaque and dockercfg types. You could check if there are any referenced secrets or pullsecrets in the service account you don't already have in the tokens list, and simply try to do a full list in the ns if there are.